content/300-guides/050-database/870-using-prisma-with-mongodb.mdx

```diff
@@ -0,0 +1,242 @@
+
+**Replica sets.**
+
+## How to migrate old data to new schemas
+
+Types:
+
+- On-demand: when you add a new field to your schema, the past data isn't touched. But when you go and read the collection, the collection is adjusted, so when you write the data back, your data now matches your schema.
+- No breaking change policy: Don't rename or delete fields, only add fields.
+- All-at-once: where you update your schema and your past data to match your new schema all at once. This is a requirement for relational databases.
+
+## How to add in missing relations after Introspection
+
+## How to filter for <inlinecode>null</inlinecode> and undefined fields
+
+To understand how MongoDB distinguishes between `null` and undefined fields, consider the example of a `User` model with an optional `name` field:
+
+```ts
+model User {
+  id    String  @id @default(auto()) @map("_id") @db.ObjectId
+  email String
+  name  String?
+}
+```
+
+First, try creating a record with the `name` field explicitly set to `null`. Prisma will return `name: null` as expected:
+
+<CodeWithResult expanded={true}>
+
+<cmd>
+
+```ts
+const createNull = await prisma.user.create({
+  data: {
+    email: 'user1@prisma.io',
+    name: null,
+  },
+})
+console.log(createNull)
+```
+
+</cmd>
+
+<cmdResult>
+
+```
+{
+    id: '6242c4ae032bc76da250b207',
+    email: 'user1@prisma.io',
+    name: null
+}
+```
+
+</cmdResult>
+
+</CodeWithResult>
+
+If you check your MongoDB database directly, you will also see a new record with `name` set to `null`:
+
+```terminal
+_id: 6242c4af032bc76da250b207
+email: "user1@prisma.io"
+name: null
+```
+
+Next, try creating a record without explicitly setting the `name` field:
+
+<CodeWithResult expanded={true}>
+
+<cmd>
+
+```ts
+  const createUndefined = await prisma.user.create({
+    data: {
+       email: "user2@prisma.io",
+     }
+   })
+   console.log(createUndefined)
+}
+```
+
+</cmd>
+
+<cmdResult>
+
+```
+{
+    id: '6242c4ae032bc76da250b208',
+    email: 'user2@prisma.io',
+    name: null
+},
+
+```
+
+</cmdResult>
+
+</CodeWithResult>
+
+Prisma still returns `name: null`, but if you look in the database directly you will see that the record has no `name` field defined at all:
+
+```terminal
+_id: 6242c4af032bc76da250b208
+email: "user2@prisma.io"
+```
+
+Prisma returns the same result in both cases, because we currently don't have a way to specify this difference in MongoDB between fields that are `null` in the underlying database, and fields that are not defined at all — see [this Github issue](https://github.com/prisma/prisma/issues/12555) for more information.
+
+This means that you currently have to be careful when searching for `null` and undefined fields. Searching for records with `name: null` will return both records:
```

Comment:
This is unfortunately the wrong way around - Prisma will return only one of the two, the one with explicit `null`. To include the other record as well you'll need to use `OR .. isSet: false` to get it too.

Reply:
Oops, thanks, that's a pretty major mistake. I searched for both at once when testing this and then muddled the order. Fixed the examples but let me know if it needs better explanation.